tool/src/gen.js

```diff
@@ -0,0 +1,152 @@
+// Usage: node tool/src/gen.js [number] > box-[number].ifc
+// Here's 1 box:
+/*
+ISO-10303-21;
+HEADER;
+FILE_DESCRIPTION(('IFC4'),'2;1');
+FILE_NAME('example.ifc','2018-08-8',(''),(''),'','','');
+FILE_SCHEMA(('IFC4'));
+ENDSEC;
+DATA;
+#100=IFCPROJECT('UUID-Project',$,'Proxy with extruded box',$,$,$,$,(#201),#301);
+#201=IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,1.0E-5,$,$);
+#202=IFCGEOMETRICREPRESENTATIONSUBCONTEXT('Body','Model',*,*,*,*,#201,$,.MODEL_VIEW.,$);
+#301=IFCUNITASSIGNMENT((#311));
+#311=IFCSIUNIT(*,.LENGTHUNIT.,$,.METRE.);
+#500=IFCBUILDING('UUID-building',$,'Test Building',$,$,#511,$,$,.ELEMENT.,$,$,$);
+#519=IFCRELAGGREGATES('UUID-RelAggregates',$,$,$,#100,(#500));
+#1000=IFCBUILDINGELEMENTPROXY('UUID-Proxy',$,'P-1','sample proxy',$,$,#1010,$,$);
+#1010=IFCPRODUCTDEFINITIONSHAPE($,$,(#1020));
+#1020=IFCSHAPEREPRESENTATION(#202,'Body','SweptSolid',(#1021));
+#1021=IFCEXTRUDEDAREASOLID(#1022,$,#1034,1.);
+#1022=IFCRECTANGLEPROFILEDEF(.AREA.,'1m x 1m rectangle',$,1.,1.);
+#1034=IFCDIRECTION((0.,0.,1.));
+#10000=IFCRELCONTAINEDINSPATIALSTRUCTURE('UUID-Spatial',$,'Physical model',$,(#1000),#500);
+ENDSEC;
+END-ISO-10303-21;
+*/
+
+
+const numBoxen = parseInt(process.argv[2] || 1)
+const projectTitle = `Project of ${numBoxen} boxes`
+
+p(`
+${stepOpen()}
+${ifcHeader(numBoxen)}
+${dataOpen(projectTitle)}
+`)
+
+boxen(numBoxen)
+
+p(`
+${dataClose()}
+${stepClose()}
+`)
+
+
+/** Prints lots of boxen */
+function boxen(num = 10) {
+  const maxSpan = 100
+  for (let i = 1; i <= num; i++) {
+    const x = parseInt(Math.random() * maxSpan)
+    const y = parseInt(Math.random() * maxSpan)
+    const z = parseInt(Math.random() * maxSpan)
+    // eslint-disable-next-line no-magic-numbers
+    const id = i * 1000
+    // eslint-disable-next-line no-magic-numbers
+    p(box(id, [x, y, z], 500, 202, 312))
```

Comment:
More a note, should these be variables at the top of the file/overridable command line params?

Reply:
Will hoist for now and maybe CLI params in future PR when using input model.